server/src/storage/mod.rs

```diff
@@ -314,6 +355,48 @@ impl StorageEngine {
     }
 }
 
+/// Log used to record changes in specific attributes in the storage engine.
+pub struct Log {
```

Comment:
Maybe include a memory or size limit on `Log`, and if possible use a circular buffer like [RRDtool](https://en.wikipedia.org/wiki/RRDtool), so it will not use an unbounded amount of memory if it is never collected. Alternative, only collect information if it will be requested, e.g., by having a command-line flag or `Action` to enable `Log`. Although this will probably add an excessive amount of complexity to save an insignificant amount of memory.

Reply:
As you mentioned it is quite minor. The intention is that the metrics are collected from all edges using a scheduled task (fx. running every 5 minutes) in the UI platform backend, which should mean that the metrics should never use too much memory. I do however think the used memory is worth considering when further designing how this scheduled task should work so I have added a link to this comment in that work item.

Reply:
My main concern is limiting the amount of memory used for collection internal metrics when the system is deployed without the UI of if the UI cannot connect to the system for an extended period of time. In this case the amount of memory used for internal metrics is unbounded and could cause out of memory errors as far as I can tell.

Reply:
That is a good point. I agree that this is something that needs to be fixed now and that adding a command-line flag would add an excessive amount of complexity. Following your other suggestion i found a [crate](https://github.com/agerasev/ringbuf) that supports circular buffers and have used it to implement a new version of the `Metric` struct. 

Currently the ring buffer capacity is set so the total memory usage of a single `Metric` should be capped at ~1 MiB. Alternatively we can change this in the future so it scales with the reserved uncompressed/compressed memory.